mikupad.html

```diff
@@ -1264,8 +1281,12 @@
 	const [tokens, setTokens] = useState(0);
 	const [predictStartTokens, setPredictStartTokens] = useState(0);
 	const [lastError, setLastError] = useState(undefined);
+	const [contextLength, setContextLength] = useSessionState('contextLength', defaultPresets.contextLength);
+	const [memoryTokens, setMemoryTokens] = useSessionState('memoryTokens', defaultPresets.memoryTokens);
+	const memoryArea = useRef();
 
 	const promptText = useMemo(() => joinPrompt(promptChunks), [promptChunks]);
+	const modifiedPrompt = useMemo(() => memoryTokens.concat(promptText.substring(Math.round(promptText.length-(contextLength*3.3) + (memoryTokens.length/3.3))+1)), [contextLength, promptText, memoryTokens]);

```

Comment:
This line is too long and complex, IMO it should be split into multiple lines for cleanness.